pkg/service/tap_service.go

```diff
@@ -27,20 +28,37 @@ func (tapService *TAPService) DisconnectFromTurbo() {
 	glog.V(4).Infof("[DisconnectFromTurbo] End *********")
 }
 
-func (tapService *TAPService) getHydraToken() (string, error) {
-	hydraToken, err := tapService.turboClient.GetHydraAccessToken()
-	if err != nil {
-		return "", err
-	}
-	return hydraToken, nil
-}
-
-func (tapService *TAPService) getJwtToken(hydraToken string) (string, error) {
-	jwtToken, err := tapService.turboClient.GetJwtToken(hydraToken)
-	if err != nil {
-		return "", err
+// get the jwtToken from the auth component using client id and client secret fed from k8s secret
+// client id and secret are not provided, empty jwtToken will be sent to the channel for websocket connection
+// empty jwtToken will be ignored in websocket
+func (tapService *TAPService) getJwtToken(refreshTokenChannel chan struct{}, jwTokenChannel chan string) {
+	for {
+		// blocked by isActive
+		<-refreshTokenChannel
```

Comment:
I wonder if we should check if the channel is closed here, and return if it is? 

```
    _, ok <-refreshTokenChannel
    if !ok {
        return
    }
```

Reply:
the ctrl-c works fine without exception, I don't see the needs of checking if channel is closed. Do you think this is necessary?